documentation/modules/configuring/proc-joining-node-pool-roles.adoc

```diff
@@ -0,0 +1,153 @@
+// Module included in the following assemblies:
+//
+// assembly-config.adoc
+
+[id='proc-joining-node-pools-roles-{context}']
+= Transitioning to dual-role nodes
+
+[role="_abstract"]
+This procedure describes how to transition from separate node pools with broker-only and controller-only roles to using a node pool with dual-roles.
+If your Kafka cluster is using node pools with dedicated controller and broker nodes, you can transition to using a single node pool with both roles.
+To do this, add the `broker` role to the controller-only node pool, rebalance the cluster to move partition replicas to the dual-role node pool, and then delete the old broker-only node pool.
+
+In this procedure, we start with two node pools `pool-a`, which has only the `controller` role and `pool-b` which has only the `broker` role:
+
+.Dual-role node pool
+[source,yaml,subs="+attributes"]
+----
+apiVersion: {KafkaNodePoolApiVersion}
+kind: KafkaNodePool
+metadata:
+  name: pool-a
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  replicas: 3
+  roles:
+    - controller
+  storage:
+    type: jbod
+    volumes:
+      - id: 0
+        type: persistent-claim
+        size: 20Gi
+        deleteClaim: false
+  # ...
+---
+apiVersion: {KafkaNodePoolApiVersion}
+kind: KafkaNodePool
+metadata:
+  name: pool-b
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  replicas: 3
+  roles:
+    - broker
+  storage:
+    type: jbod
+    volumes:
+      - id: 0
+        type: persistent-claim
+        size: 20Gi
+        deleteClaim: false
+  # ...
+----
+
+The Kafka cluster has six nodes:
+
+.Kafka nodes in the node pools
+[source,shell]
+----
+NAME                 READY  STATUS   RESTARTS
+my-cluster-pool-a-0  1/1    Running  0
+my-cluster-pool-a-1  1/1    Running  0
+my-cluster-pool-a-2  1/1    Running  0
+my-cluster-pool-b-3  1/1    Running  0
+my-cluster-pool-b-4  1/1    Running  0
+my-cluster-pool-b-5  1/1    Running  0
+----
+
+The `pool-a` nodes perform the role of controller.
+The `pool-b` nodes perform the role of broker.
+
+NOTE: During this process, the ID of the node that holds the partition replicas changes. Consider any dependencies that reference the node ID.
+
+.Prerequisites
+
+* xref:deploying-cluster-operator-str[The Cluster Operator must be deployed.]
+* xref:proc-configuring-deploying-cruise-control-str[Cruise Control is deployed with Kafka.]
+
+.Procedure
+
+. Edit the node pool `pool-a` and add the `broker` role to it.
++
+.Example node pool configuration
+[source,yaml,subs="+attributes"]
+----
+apiVersion: {KafkaNodePoolApiVersion}
+kind: KafkaNodePool
+metadata:
+  name: pool-b
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  replicas: 3
+  roles:
+    - controller
+    - broker
+  storage:
+    type: jbod
+    volumes:
+      - id: 0
+        type: persistent-claim
+        size: 100Gi
+        deleteClaim: false
+  # ...
+----
+
+. Apply the new `KafkaNodePool` resource to add the `broker` role to the `pool-a` nodes.
```

Comment:
We could add an example of the command for this step as well if you want.

Reply:
The general expectation is that every user does it differently. Someone does `kubectl edit`, someone does `kubectl apply` on a YAML file. Someone would need to commit the change to Git to have it propagated. I removed the Apply part completely as it might be confusing.